bundles/http/Citrix/SAP-Ariba/Ariba_v2.js

```diff
@@ -1,399 +1,410 @@
+    return groups
 }
 
-async function addIfMissingGroup(sync, groupId, groups) {
-	groups.ids.push(groupId) //if group not found then we try to ask ariba
-	let responseData = await sync.client.fetch('groups/' + groupId + '/members?realm=' + sync.integrationParameters.realm, {
-		headers: {
-			'apikey': sync.integrationParameters.apiKey
-		}
-	})
-	if (responseData.ok) {
-		let memberList = await responseData.json()
-		for (let member of memberList) {
-			member.group = groupId
-		}
-		groups.members = groups.members.concat(memberList)
-	}
-	else {
-		console.log('Warning: unable to get Group with ID:' + groupId + ':' + responseData.statusText)
-	}
-	return groups
+async function resolveGroupMembers(sync, groupId, groups) {
+    if (groups.has(groupId)) {
+       // console.log(`Found group ${groupId} in groups cache`);
+        return groups.get(groupId);
+    }
+    let responseData = await sync.client.fetch('groups/' + groupId + '/members?realm=' + sync.integrationParameters.realm + '&limit=1000', {
+        headers: {
+            'apikey': sync.integrationParameters.apiKey
+        }
+    });
+
+    if (responseData.ok) {
+        let memberList = (await responseData.json()).map(member => {
+            return {
+                "emailAddress": member.emailAddress,
+                "phone": member.phone,
+                "name": member.name,
+                "uniqueName": member.uniqueName,
+                "passwordAdapter": member.passwordAdapter,
+            }
+        })
+        groups.set(groupId, memberList);
+        //console.log(JSON.stringify(memberList));
+
+        console.log(`Adding group ${groupId} with member list size ${memberList.length}, total group count: ${groups.size}`);
+        return memberList;
+    } else {
+        console.log('Warning: unable to get Group with ID:' + groupId + ':' + responseData.statusText);
+        return [];
+    }
+
 }
 
-function resolveGroupApproval(json, groups) { //currently we are able to store only flat json object, so all nested object must be flattened
-	const groupId = json.id
-	let userList = groups.members.filter(function(groupMember) {
-		return groupMember.group == groupId
-	})
-	for (let user of userList) { //creating list of approvers with additionl data from requisition
-		try {
-			user.approvalRequestId = json.approvalRequestId
-			user.approvalRequired = json.approvalRequired
-			user.approvalState = json.approvalState
-			user.requisitionId = json.requisitionId
-			user.reason = json.reason
-			user.state = json.state
-			user.totalCostAmount = json.totalCostAmount
-			user.totalCostCurrency = json.totalCostCurrency
-		}
-		catch (e) {
-			console.log('Warning: unable to get all requisition data for user:' + user)
-		}
-	}
-	return userList
+function resolveGroupApproval(json, groupMembers, groupId) {
+    //currently we are able to store only flat json object, so all nested object must be flattened
+    return groupMembers.map(member => {
```

Comment:
the same tip as for `resolveGroupMembers`

Reply:
Will not update